autopairs.sh

```diff
@@ -9,20 +9,23 @@ function __autopair() {
   local opening_char="$2"
   local closing_char="$3"
   local cursor_char="${READLINE_LINE:READLINE_POINT:1}"
+  local previous_char="${READLINE_LINE:READLINE_POINT-1:1}"
   local num_of_char
 
   local s
   s="${READLINE_LINE::READLINE_POINT}"
 
-  if [[ "$opening_char" == "$closing_char" ]]; then
-    num_of_char="${READLINE_LINE//[^$typed_char]/}"
-    num_of_char="${#num_of_char}"
+  num_of_char="${READLINE_LINE//\\${typed_char}}"
```

Comment:
You don't need the brackets around the variable name.

```suggestion
  num_of_char="${READLINE_LINE//\\$typed_char}"
```

Reply:
will remove it. I just thought that since there is another character included,  then variable must be in braces for readability